internal/api/token.go

```diff
@@ -159,6 +167,8 @@ func (a *API) Token(w http.ResponseWriter, r *http.Request) error {
 	grantType := r.FormValue("grant_type")
 
 	switch grantType {
+	case "pkce":
```

Comment:
hmm why would the `grantType` be pkce here? i would imagine it to be a query param that's added to the existing grant types (for `password` & `refresh_token`)

Reply:
Maybe it might be clearer to split this one out into a separate endpoint like `/oauth/token` since this represents the step to verify the authorization code with the code_verifier 

The other grant types in `/token` represent an earlier part of the oauth flow where the user presents a set of credentials to obtain an authorization code.

Reply:
Let's move to /token as it seems to be the convention